### ISSUE TITLE
Eclipse UI Tests log seem to include useless setup / teardown output

### ISSUE THREAD

**Issue by vogella** [eclipse-platform/eclipse.platform.ui#1229]
Looking at the Eclipse UI Tests build logs I see lots of:

```
----- testGetDescriptor
testGetDescriptor: setUp...
testGetDescriptor: tearDown...
```
Is this useful? I typically not interested in lot of output in case tests are running fine, but more interested in output if something goes wrong. This makes it harder in my opinion to find the relevant output.

WDYT @fedejeanne and @HeikoKlare? Is this useful output for you (as you are currently working on the tests).

**Comment by iloveeclipse** [comment]
> Is this useful?

Yes, if the test start to hang / log some not properly handled exceptions you will see which one is behaving wrong.

**Comment by vogella** [comment]
@iloveeclipse for this, I assume the listing of the test name would be sufficient.

**Comment by HeikoKlare** [comment]
In general, I would not consider this log output useful. Just logging the test name to see which tests hangs should be sufficient and even then there should, in the best case, rather be a timeout mechanism that aborts the tests and logs the according information about the hanging test anyway, without the need for the test to always print it on its own.

And still logging the test name is only useful as parallel builds are not enabled yet. In case of having parallel builds, you need to somehow attach the information about the test to the log output (such as an unhandled exception) as an exception could basically belong to any of the test projects that are currently built.

But maybe there are other use cases for the output that I do not see yet due to the comparably short time I have been working on test failures.

**Comment by fedejeanne** [comment]
> WDYT @fedejeanne and @HeikoKlare? Is this useful output for you (as you are currently working on the tests).

Well it does clutter the output a bit but it is indeed the easiest way to see which test is currently hanging in "real time". Regarding the last part: I also see that there is some duration information after all tests ran in the log _e.g._ `org.eclipse.ui.tests.api.IPageLayoutTest.testGetDescriptor -- Time elapsed: 0.780 s`, but this is only added **after** all tests ran and I'm not sure whether or not it's accurate (I've seen that some tests run in < 1s even though the "live console" of the CI build takes a lot of time to log the next call to the `setUp` method). 

@iloveeclipse taking all this into consideration, do you think this kind of log information (`... -- Time elapsed: 0.780 s)` would still be good enough to see which test is delaying an execution _i.e._ which test should be optimized?

**Comment by iloveeclipse** [comment]
I've never had a problem with setup/teardown output and that helped me to diagnose sporadic test hangs. That's all what I can say here.

I don't understand the problem we are trying to fix, so feel free to do whatever you want, but next time tests hang I would expect the committer that will wish no output will also go and investigate the problem.

**Comment by fedejeanne** [comment]
I think we need a new logging concept that allows for finer logging ("debug") to be enabled under demand. That would de-clutter the normal logging output ("info") and would also allow us to enable the finer logging and look for errors if necessary.

I'm thinking of Log4J/LogBack, the problem is that I'm not sure how to easily change the logging level in the ci-build so that would be a no-go.

Can you guys think of a better option?

**Comment by HeikoKlare** [comment]
The problem with on-demand logging is that in case it is not activated for CI builds, you will not have the logging for random/sporadic failures/hangs. And reproduction will usually be quite hard. So I see the points of @iloveeclipse. So we should probably not remove those outputs without having a proper/better alternative.

Concerning test output, I'd propose to move away from printing to stdout and use a logging framework instead. As mentioned, in case we switch to parallel builds at some point in time, the direct console outputs without containing information about the context, i.e., the test in which it occurred, render rather useless.

For hangs that do not produce any output, however, shouldn't we have some detection mechanism anyway? Something like every test case spawing a thread that monitors whether the test took too long (like the UI freeze monitor), so that this thread can properly log a hanging test?

**Comment by fedejeanne** [comment]
> I'd propose to move away from printing to stdout and use a logging framework instead

I agree. Is **Log4J** an option these days or did it get banned after the Log4Shell incident? I'm asking because I haven't seen any single reference to it in my workspace and it's not in the TP either, which is odd.

Coming back to the original question from @vogella : IMO it's only useful in a very small set of cases and there might be a way to use on-demand logging and de-clutter a bit by using parameters (see [this](https://stackoverflow.com/questions/62736315/invoke-github-actions-workflow-manually-and-pass-parameters/62736541#62736541)) but I tried it in my fork and it didn't work, so I can't provide a PR for that.

**Comment by HeikoKlare** [comment]
> Coming back to the original question from @vogella : IMO it's only useful in a very small set of cases and there might be a way to use on-demand logging and de-clutter a bit by using parameters (see [this](https://stackoverflow.com/questions/62736315/invoke-github-actions-workflow-manually-and-pass-parameters/62736541#62736541)) but I tried it in my fork and it didn't work, so I can't provide a PR for that.

The problem is that you do sometimes only know that you require more detailed logging when it is already too late, i.e., when the test got stuck. On-demand logging is good if you have reproducible errors, so you can just run the build again with logging enabled. But there have tests been stuck only once every few weeks or even months. On-demand logging won't help you there. But as mentioned, some watchdog functionality might be a solution for that case.
